lib/pages/chat/chat.dart

```diff
@@ -659,60 +662,61 @@ class ChatController extends State<Chat>
     );
   }
 
-  void voiceMessageAction() async {
-    // ignore: unused_local_variable
-    final scaffoldMessenger = ScaffoldMessenger.of(context);
-    if (PlatformInfos.isAndroid) {
-      final info = await DeviceInfoPlugin().androidInfo;
-      if (info.version.sdkInt < 19) {
-        showOkAlertDialog(
-          context: context,
-          title: L10n.of(context)!.unsupportedAndroidVersion,
-          message: L10n.of(context)!.unsupportedAndroidVersionLong,
-          okLabel: L10n.of(context)!.close,
-        );
-        return;
-      }
-    }
-
-    if (await Record().hasPermission() == false) return;
-    final result = await showDialog<RecordingResult>(
-      context: context,
-      useRootNavigator: false,
-      barrierDismissible: false,
-      builder: (c) => const RecordingDialog(),
+  Future<void> sendVoiceMessageAction({
+    required TwakeAudioFile audioFile,
+    required Duration time,
+    required List<int> waveform,
+  }) async {
+    final fileInfo = FileInfo(
+      audioFile.name,
+      audioFile.filePath ?? '',
```

Comment:
please check filePath here and show toast in this case: `Audio message failed to send.`